src/cyverse_groups_client/core.clj

```diff
@@ -136,15 +154,19 @@
                      {:query-params {:user user}
                       :as           :json})))
 
-  (revoke-folder-privilege [_ user name subject privilege]
-    (:body (http/delete (build-url base-url "folders" name "privileges" subject privilege)
-                        {:query-params {:user user}
-                         :as           :json})))
-
-  (grant-folder-privilege [_ user name subject privilege]
-    (:body (http/put (build-url base-url "folders" name "privileges" subject privilege)
-                     {:query-params {:user user}
-                      :as           :json})))
+  (revoke-folder-privilege [_ user folder-name subject privilege]
+    (let [privilege (name privilege)]
+      (validate-folder-privilege privilege)
+      (:body (http/delete (build-url base-url "folders" folder-name "privileges" subject privilege)
+                          {:query-params {:user user}
+                           :as           :json}))))
+
+  (grant-folder-privilege [_ user folder-name subject privilege]
+    (let [privilege (name privilege)]
+      (validate-folder-privilege privilege)
```

Comment:
I'd originally planned to pass strings in for this argument, but when I added the validation, keywords seemed like a natural fit. I ended up deciding to support both for convenience, so converting the string to a keyword in the validation function was required to ensure that the privilege was a keyword before attempting to do the validation. The second problem came up when I tried to pass a keyword into `build-url` because it was encoding the leading colon in the keyword, so I added the conversion in the URL. A better choice would probably have been to put the conversion in `build-url`, however. I'll do that.